trigger/build.go

```diff
@@ -30,21 +30,17 @@ func (t *buildTriggers) taskStatusToDesc() string {
 	other := 0
 	noReport := 0
 	for _, t := range t.tasks {
-		switch t.Status {
-		case evergreen.TaskSucceeded:
+		switch {
+		case t.Status == evergreen.TaskSucceeded:
 			success++
 
-		case evergreen.TaskFailed:
+		case t.Status == evergreen.TaskFailed:
 			failed++
 
-		case evergreen.TaskSystemFailed, evergreen.TaskTimedOut,
-			evergreen.TaskSystemUnresponse, evergreen.TaskSystemTimedOut,
-			evergreen.TaskTestTimedOut:
+		case utility.StringSliceContains(evergreen.TaskNonGenericFailureStatuses, t.Status):
```

Comment:
This case is specifically system failure statuses; TaskNonGenericFailureStatuses includes TaskSetupFailure